codice/alliance
# Cal 10 - Create Nitf Input Transformer

#### What does this PR do?
CAL-10 Create NITF Transformer  …
Removed the Nitf constants interface, replacing it with NitfHeaderAttribute, GraphicsAttribute, ImageAttribute, TextAttribute.  Removed the XML metadata block from the metacard.  Added attributes for all of the available NITF fields and added support for multiple image segments, graphic segments and text segments.  Created catalog-imaging-app so the transformer can be loaded as a karaf application.

#### Who is reviewing it?
@beyelerb 
@kcwire (hero)
@jlcsmith 

#### How should this be tested?
1) pull and build https://github.com/codice/ddf/pull/633.  
2) start DDF.
3) install the Alliance Imaging application.
3) ingest a NITF file.
4) execute a search and verify that the results contain the ingested NITF.
5) switch to the Details tab to see the extended NITF properties.

#### Any background context you want to provide?
This PR can be merged before or after https://github.com/codice/ddf/pull/633.  However, it won't store the extended NITF properties until after that PR has been merged.

#### What are the relevant tickets?
CAL-10

#### Screenshots (if appropriate)
N/A

#### Checklist:
